blockchain/pallets/minting/src/tests.rs

```diff
@@ -338,5 +340,39 @@ mod register_identity {
                 );
             });
         }
+
+        #[test]
+        fn register_for_minting_is_free_for_first_call() {
+            use frame_support::pallet_prelude::Pays;
+
+            fn gen_tree(r: &[&str]) -> MerkleTree<Blake2b> {
+                MerkleTree::from_iter(r).unwrap()
+            }
+
+            new_test_ext().execute_with(|| {
+                register_id_account(42, 1);
+
+                // first call is free, actual_weight set to 0
+                let tree_0 = gen_tree(&["a", "b"]);
+                assert_eq!(
+                    FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_0),
+                    Ok(PostDispatchInfo {
+                        actual_weight: Some(0),
+                        pays_fee: Pays::Yes
```

Comment:
Does it not work if the first PostDispatchInfo specifies Pays::No and the second one Pays::Yes?